javav2/example_code/s3/src/main/java/com/example/s3/express/S3DirectoriesScenario.java

```diff
@@ -0,0 +1,578 @@
+import software.amazon.awssdk.services.iam.model.CreateAccessKeyResponse;
+import software.amazon.awssdk.services.iam.model.IamException;
+import software.amazon.awssdk.services.s3.S3AsyncClient;
+import software.amazon.awssdk.services.s3.model.S3Exception;
+
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Scanner;
+import java.util.UUID;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.CompletionException;
+import java.util.stream.IntStream;
+
+public class S3DirectoriesScenario {
+
+    public static final String DASHES = new String(new char[80]).replace("\0", "-");
+
+    private static final Logger logger = LoggerFactory.getLogger(S3DirectoriesScenario.class);
+    static Scanner scanner = new Scanner(System.in);
+
+    private static S3AsyncClient mS3RegularClient;
+    private static S3AsyncClient mS3ExpressClient;
+
+    private static String mdirectoryBucketName;
+    private static String mregularBucketName;
+
+    private static String stackName = "cfn-stack-s3-express-basics--" + UUID.randomUUID();
+
+    private static String regularUser = "";
+    private static String vpcId = "";
+    private static String expressUser = "";
+
+    private static String vpcEndpointId = "";
+
+    private static final S3DirectoriesActions s3DirectoriesActions = new S3DirectoriesActions();
+
+    public static void main(String[] args) {
+        try {
+            s3ExpressScenario();
+        } catch (RuntimeException e) {
+            logger.info(e.getMessage());
+        }
+    }
+
+    // Runs the scenario.
+    private static void s3ExpressScenario() {
+        logger.info(DASHES);
+        logger.info("Welcome to the Amazon S3 Express Basics demo using AWS SDK for Java V2");
+        logger.info("""
+            Let's get started! First, please note that S3 Express One Zone works best when working within the AWS infrastructure,
+            specifically when working in the same Availability Zone. To see the best results in this example and when you implement
+            Directory buckets into your infrastructure, it is best to put your compute resources in the same AZ as your Directory
+            bucket.
+            """);
+        waitForInputToContinue(scanner);
+        logger.info(DASHES);
+
+        // Create an optional VPC and create 2 IAM users.
+        UserNames userNames = createVpcUsers();
+        String expressUserName = userNames.getExpressUserName();
+        String regularUserName = userNames.getRegularUserName();
+
+        //  Set up two S3 clients, one regular and one express,
+        //  and two buckets, one regular and one express.
+        setupClientsAndBuckets(expressUserName, regularUserName);
+
+        // Create an S3 session for the express S3 client and add objects to the buckets.
+        logger.info("Create an S3 session for the express S3 client and add objects to the buckets");
```

Comment:
Add information about an S3 session: what happens in a session? what operations require a session?